packages/components/src/components/input/input.stories.ts

```diff
@@ -0,0 +1,675 @@
+ */
+
+export const Parts = {
+  parameters: {
+    controls: {
+      exclude: ['label', 'left', 'right', 'clear-icon', 'help-text', 'clearable', 'value']
+    }
+  },
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: {
+          type: 'template',
+          name: 'sd-input::part(...){outline: solid 2px red}',
+          values: [
+            'form-control',
+            'form-control-label',
+            'form-control-input',
+            'form-control-help-text',
+            'base',
+            'border',
+            'input',
+            'left',
+            'clear-button',
+            'right'
+          ].map(part => {
+            return {
+              title: part,
+              value: `<style>#part-${part} sd-input::part(${part}){outline: solid 2px red}</style><div id="part-${part}">%TEMPLATE%</div>`
+            };
+          })
+        }
+      },
+      constants: [
+        { type: 'attribute', name: 'clearable', value: true },
+        { type: 'attribute', name: 'value', value: 'value' },
+        { type: 'attribute', name: 'label', value: 'Label' },
+        { type: 'attribute', name: 'help-text', value: 'help-text' },
+        {
+          type: 'slot',
+          name: 'left',
+          value: '<sd-icon slot="left" library="global-resources" name="system/picture"></sd-icon>'
+        },
+        {
+          type: 'slot',
+          name: 'right',
+          value: '<sd-icon slot="right" library="global-resources" name="system/picture"></sd-icon>'
+        }
+      ],
+      args
+    });
+  }
+};
+
+/**
+ * `sd-input` is fully accessibile via keyboard.
+ */
+
+export const Mouseless = {
+  render: (args: any) => {
+    return html`<div class="mouseless w-[231px]">${generateTemplate({ args })}</div>`;
+  },
+
+  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
+    const el = canvasElement.querySelector('.mouseless sd-input');
+    await waitUntil(() => el?.shadowRoot?.querySelector('input'));
+    el?.shadowRoot?.querySelector('input')!.focus();
+  }
+};
+
+/**
+ * Step function used in Stepper sample.
+ */
+const step = (direction: 'up' | 'down') => {
+  // Get input element and buttons
+  const inputEl = document.querySelector('sd-input')!;
+  const stepDownButton = document.querySelector('#stepDownButton')!;
+
+  // function to add class if not present
+  const addClass = (el: Element, className: string) => {
+    if (!el.classList.contains(className)) {
+      el.classList.add(className);
+    }
+  };
+
+  // function to remove class if present
+  const removeClass = (el: Element, className: string) => {
+    if (el.classList.contains(className)) {
+      el.classList.remove(className);
+    }
+  };
+
+  // function to adjust classlist of stepDownButton
+  const adjustClasslist = () => {
+    if (inputEl.value === '0') {
+      addClass(stepDownButton, 'text-neutral-500');
+    } else {
+      removeClass(stepDownButton, 'text-neutral-500');
+    }
+  };
```

Comment:
The problem for the docs might be, that this is maybe not available inside the story.

Could you insert all this JS inside the story? This is especially important as people should be able to see, what you're doing and if you have it outside the story that doesn't work. :)